api/controllers/pack.js

```diff
@@ -177,4 +177,66 @@ router.post('/remove-item', authenticate, (req, res) => {
 
 });
 
+// make a copy of a pack and its items
+router.post('/copy-pack', authenticate, (req, res) => {
+    const { packId } = req.body;
+
+    // verify ownership
+    if (packId) {
+        models.Pack.findOne({ where: { id: packId, userId: req.user.id } })
+            .then(pack => {
+                if (!pack) {
+                    return res.sendStatus(401);
+                }
+                else { //owner of pack is making request
+                        models.Pack.findOne({
+                            where: { id: packId },
+                            include: [
+                                {
+                                model: models.Item, through: { where: { packId: packId } }, include: [
+                                    { model: models.Category, as: 'Category', attributes: ['name'] }
+                                ]
+                                },
+                            ]
+                        })
+                        .then(pack => {
+                            models.Pack.create({ userId: req.user.id,
+                                                title: "Copy of " + pack.title,
+                                                public: pack.public,
+                                                description: pack.description,
+                                                duration: pack.duration,
+                                                duration_unit: pack.duration_unit,
+                                                temp_range: pack.temp_range,
+                                                season: pack.season,
+                                                gender: pack.gender
+                                                })
```

Comment:
This can be condensed down to:
`models.Pack.create({ ...pack, userId: req.user.id, title: "Copy of " + pack.title })`

Using `...pack`, you're copying all of the properties to the new pack and then simply overriding the new properties, e.g. userId and title.

Reply:
Ah, that's super cool. I'd never seen that before until I started looking at Packstack. 

Should be all good now, I just pushed the change.